ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/PostCommitValidation.scala

```diff
@@ -282,33 +289,61 @@ private[appendonlydao] object PostCommitValidation {
 
     def toStateV1RejectionReason: v1.RejectionReason
 
-    def toStateV2RejectionReason: v2.Update.CommandRejected.RejectionReasonTemplate
+    def toStateV2RejectionReason(errorFactories: ErrorFactories)(implicit
+        contextualizedErrorLogger: ContextualizedErrorLogger
+    ): v2.Update.CommandRejected.RejectionReasonTemplate
   }
 
   object Rejection {
 
     import com.daml.platform.store.Conversions.RejectionReasonOps
-
     object UnknownContract extends Rejection {
+      override val description = "Unknown contract"
+
+      override def toStateV1RejectionReason: v1.RejectionReason =
+        v1.RejectionReasonV0.Inconsistent(description)
+
+      override def toStateV2RejectionReason(errorFactories: ErrorFactories)(implicit
+          contextualizedErrorLogger: ContextualizedErrorLogger
+      ): v2.Update.CommandRejected.RejectionReasonTemplate =
+        domain.RejectionReason
+          .Inconsistent(description)
+          .toParticipantStateRejectionReason(ErrorFactories(new ErrorCodesVersionSwitcher(false)))(
+            NoLogging
+          )
+    }
+
+    final case class UnknownContracts(
```

Comment:
What's the difference between `UnknownContracts` and `UnknownContract`?

Reply:
`UnkownContract` is the legacy answer for the failure. I'll find a better name and reason for it

Reply:
Renamed it to `MaximumLedgerTimeLookupFailure` 87c9b918072712fed1b6224060a2b39f34cf6feb